packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,485 @@
+          selectors.initializationError
+        );
+
+        expect(initializationError).not.toBeNull();
+        expect(console.error).toHaveBeenCalledTimes(1);
+        expect(console.error).toHaveBeenCalledWith(expectedConsoleErrorMessage);
+        consoleErrorSpy.mockRestore();
+      });
+    });
+  });
+
+  describe('when an initialization error occurs', () => {
+    beforeEach(() => {
+      mockErroneousHeadlessInitialization();
+    });
+
+    it('should display the error component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.initializationError
+      );
+
+      expect(initializationError).not.toBeNull();
+    });
+  });
+
+  describe('when the component is initialized successfully', () => {
+    beforeEach(() => {
+      mockSuccessfulHeadlessInitialization();
+      prepareHeadlessState();
+    });
+
+    describe('when there are no recommendations', () => {
+      it('should not display a recommendation list', async () => {
+        recommendationListState = {
+          isLoading: false,
+          recommendations: [],
+          error: null,
+          searchResponseId: null,
+        };
+        const element = createTestComponent();
+        await flushPromises();
+
+        const recommendationContainer = element.shadowRoot.querySelector(
+          selectors.gridContainer
+        );
+
+        expect(recommendationContainer).toBeNull();
+        const recommendationListItems = element.shadowRoot.querySelectorAll(
+          selectors.recommendationListItem
+        );
+        expect(recommendationListItems.length).toBe(0);
+      });
+    });
+
+    describe('when there are recommendations', () => {
+      describe('when the variant value is #grid', () => {
+        it('should properly display the recommendations as a grid', async () => {
+          const element = createTestComponent({
+            ...defaultOptions,
+            variant: 'grid',
+          });
+          await flushPromises();
+
+          const recommendationContainer = element.shadowRoot.querySelector(
+            selectors.gridContainer
+          );
+          const recommendationItems = element.shadowRoot.querySelectorAll(
+            selectors.recommendationListResult
+          );
+
+          expect(recommendationContainer).not.toBeNull();
+          expect(recommendationItems.length).toBe(recommendationsArray.length);
+          recommendationItems.forEach((item, index) => {
+            expect(item.result.title).toEqual(
+              recommendationsArray[index].title
+            );
+            expect(item.result.clickUri).toEqual(
+              recommendationsArray[index].clickUri
+            );
+            expect(item.result.excerpt).toEqual(
+              recommendationsArray[index].excerpt
+            );
+            expect(item.engineId).toBe(defaultOptions.engineId);
```

Comment:
we can do a deep equality to test everything in one line:
```
expect(item.result).toEqual(recommendationsArray[index]);
expect(item.engineId).toBe(defaultOptions.engineId);
```
you can even add a check to check that resultTemplatesManager was passed properly.